src/shims/posix/linux/foreign_items.rs

```diff
@@ -182,7 +182,15 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     id if id == sys_futex => {
                         futex(this, args, dest)?;
                     }
-                    id => throw_unsup_format!("Miri does not support syscall ID {}", id),
+                    id => {
+                        if this.eval_context_ref().machine.panic_on_unsupported_syscalls {
+                            // message is slightly different here to make automated analysis easier
+                            this.start_panic(format!("unsupported Miri functionality: syscall ID {} is not emulated", id).as_ref(), None)?;
```

Comment:
I see you choose a different message than `throw_unsup_format!`... would it make sense to always use the same message for both? That would also mean we don't have to bikeshed the details of the messages. ;)